liquidprompt

```diff
@@ -2971,6 +2994,48 @@ _lp_git_staged_lines() {
     lp_vcs_staged_d_lines=$lp_git_diff_shortstat_d_lines
 }
 
+# Get the remote name for the current branch.
+# NOTE: Assumes that `_lp_find_vcs` and `_lp_vcs_branch` (costly functions) have been called.
+_lp_git_remote() {
+    lp_vcs_remote=""
+
+    # If there is no branch…
+    if [[ -z "$lp_vcs_branch" ]]; then
+        # … then there is no remote.
+        return 1
+    fi
+
+    # If we don't have a VCS root…
+    if [[ -z "$lp_vcs_root" ]]; then
+        # … we cannot find the config file.
+        return 1
+    fi
+
+    local file="${lp_vcs_root}/.git/config"
```

Comment:
I just realized, this is a big problem.

I made this mistake before. Not all Git directories have a `.git/` directory. Some have a `.git` file. See https://git-scm.com/docs/git-worktree for details.

See commit fc44d9724127e6b3bd803be5afed45d64fa566e4 for the fix I made to add this support (after I broke it in the first place), and 1be52e0d048f68db9ab0ffd2f4d2a4212b685f93 for more details.

The point is `$lp_vcs_root` is not a safe way to get to the config file. What we should do instead is use `$lp_vcs_dir`, which is supposed to be the database directory.

I say "supposed to be", because it turns out the `GIT_DIR` for a worktree is actually the `.git/worktrees/<worktree_name>/`.

The problem is we cannot "fix" our detection of `$lp_vcs_dir` to always resolve to the `.git/` directory, because some functions like `_lp_git_head_status()` need to 
read files like `REBASE_HEAD` that exist in the worktree dir.

That said, this seems to be not the only bug of this sort:
https://github.com/liquidprompt/liquidprompt/blob/b0af1a5eff5da520fcb02842aea35caef7e890ad/liquidprompt#L2571
The Git SVN detection is likely broken in Git worktrees.

I think the right solution is to have two Git dirs, the "real" one, and the "working", or "specific" one. Then we can load files from which ever is correct. The trick will be how do we detect the "real" one from a worktree? It looks like a file named `gitdir` exists in the "specific" git dir that contains the path to the "real" one, so we can probably load that.

That was a lot. And this is pretty complicated (and out of scope for this PR), so I can implement it.

Reply:
What about going back at your first suggested option of just calling `git remote`? Letting Git managing its complex configuration seems a better option given all this.

Reply:
Yeah, that is an option. Since there is already a bug in Git SVN detection though, I'm thinking that I need to fix that, and once I do that, this should be pretty simple to still do this way.